problem14.rb

```diff
@@ -0,0 +1,43 @@
+#problem 14 project euler
+
+
+def collatz(x)
+    has2={}
+    seq=[]
+    seq<<x
+    temp=x
+    while(temp>1)
+        if temp%2==0
+            temp=(temp/2).to_i
+            if has2.include? temp
+                seq+=has2[temp]
+                break
+            else
+                seq<<temp
+            end    
+        else
+            temp=3*temp+1
+            if has2.include? temp
+                seq+=has2[temp]
+                break
+            else
+                seq<<temp
+
+            end
+        end
+    end
+    has2[x]=seq        
+    return seq.length()
```

Comment:
```suggestion
    seq.size
```
Нет смысла писать пустые скобки